packages/backend-modules/mail/NewsletterSubscription.js

```diff
@@ -1,9 +1,8 @@
 const createNewsletterSubscription = (interestConfigurationMap) => ({
-  buildSubscription(userId, interestId, subscribed) {
-    const interestConfiguration = this.interestConfiguration(interestId)
-    const name = interestConfiguration.name
+  buildSubscription(userId, interestId, subscribed, roles) {
+    const { name, ...rest } = this.interestConfiguration(interestId)
     const id = Buffer.from(userId + name).toString('base64')
-    return { name, id, subscribed }
+    return { userId, subscribed, roles, name, ...rest, id }
```

Comment:
sylistically speaking, I'd put `...rest` at the end of the enumeration.

Reply:
I usually agree. However, in this instance, I do want anything in `...rest` to be overwritten by other variables. (Did put `...rest` at begining now.)